internal/testrunner/runners/system/servicedeployer/factory.go

```diff
@@ -5,27 +5,66 @@
 package servicedeployer
 
 import (
-	"errors"
 	"os"
 	"path/filepath"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/logger"
 )
 
+const devDeployDir = "_dev/deploy"
+
 var (
 	// ErrNotFound is returned when the appropriate service runner for a package
 	// cannot be found.
 	ErrNotFound = errors.New("unable to find service runner")
 )
 
-// Factory chooses the appropriate service runner for the given package, depending
-// on service configuration files defined in the package.
-func Factory(packageRootPath string) (ServiceDeployer, error) {
-	packageDevPath := filepath.Join(packageRootPath, "_dev")
+// FactoryOptions defines options used to create an instance of a service deployer.
+type FactoryOptions struct {
+	PackageRootPath    string
+	DataStreamRootPath string
+}
+
+// Factory chooses the appropriate service runner for the given data stream, depending
+// on service configuration files defined in the package or data stream.
+func Factory(options FactoryOptions) (ServiceDeployer, error) {
+	devDeployPath, err := findDevDeployPath(options)
+	if err != nil {
+		logger.Errorf("can't find _dev/deploy directory")
```

Comment:
You're defining `devDeployDir` as a constant earlier. Might as well use it here. 🙂

Reply:
Fixed